common/decklist.cpp

```diff
@@ -463,32 +465,84 @@ bool DeckList::saveToFile_Native(QIODevice *device)
     return true;
 }
 
+void DeckList::removeEmptyLinesFromStart(QVector<QString> &inputs)
+{
+    // https://stackoverflow.com/questions/4642339/trimming-a-vector-of-strings
+    inputs.erase(inputs.begin(), std::find_if(inputs.begin(), inputs.end(), [](const QString& s) {return !s.isEmpty();}));
+}
+
 bool DeckList::loadFromStream_Plain(QTextStream &in)
 {
     cleanList();
+    bool inSideboard = false, titleFound = false, isSideboard;
+    int okRows = 0, blankLines = 0;
+    QVector<QString> inputs;
+
+    // Lets convert input QTextStream => QVector
+    while (!in.atEnd())
+        inputs.push_back(in.readLine().simplified().toLower());
+
+    // "Trim" the inputs of all blank lines at the start & end of paste
+    removeEmptyLinesFromStart(inputs);
+    std::reverse(inputs.begin(), inputs.end()); // Reverse vector (to remove blanks from end)
+    removeEmptyLinesFromStart(inputs);
+    std::reverse(inputs.begin(), inputs.end()); // Vector now in correct order
+
+    // This will "trim" multiple blank lines in a row to just one blank
+    // Ex: ("Card1", "Card2", "", "", "", "Card3") => ("Card1", "Card2", "", "Card3")
+    while (true) {
+        auto i1 = std::adjacent_find(inputs.begin(), inputs.end());
+
+        if (i1 != inputs.end()) {
+            if (i1 != QString()) {
+                // So we have a duplicate row like (..., "1 card1", "3 card1", ...)
+                // To deal with this, we will combine these rows together to form (..., "4 card1", ...)
+                // And replace the newest row (aka we'd edit "3 card1" in our example) with the new value
+                int combinedTotal = (*i1).mid(0,1).toInt() + (*i1+1).mid(0,1).toInt();
+                inputs.replace(static_cast<int>(std::distance(inputs.begin(), i1+1)), QString::number(combinedTotal) + (*(i1+1)).mid(1));
```

Comment:
Is there a good reason to do this? From the comment above this loop was meant to only delete double empty lines.
Also, what happens if my deck does't contain card numbers, or has more that 9 cards (that is, a number with more than a single digit) with the same name?
Example 1:
```
Kitten
Kitten
Kitten
Kitten
```

Example 2:
```
12 Kitten
13 Kitten
```